core/merkle.py

```
@@ -0,0 +1,102 @@
+import hashlib
+import json
+from typing import List, Optional, Tuple
```

Ruff (0.15.1)

[warning] `typing.List` is deprecated, use `list` instead

(UP035)